usermods/audioreactive/audio_reactive.h

```diff
@@ -638,13 +645,11 @@ class AudioReactive : public Usermod {
 #ifdef ARDUINO_ARCH_ESP32
     // used for AGC
     int      last_soundAgc = -1;   // used to detect AGC mode change (for resetting AGC internal error buffers)
-    double   control_integrated = 0.0;   // persistent across calls to agcAvg(); "integrator control" = accumulated error
-
-
+    float    control_integrated = 0.0f;   // persistent across calls to agcAvg(); "integrator control" = accumulated error
     // variables used by getSample() and agcAvg()
     int16_t  micIn = 0;           // Current sample starts with negative values and large values, which is why it's 16 bit signed
-    double   sampleMax = 0.0;     // Max sample over a few seconds. Needed for AGC controller.
-    double   micLev = 0.0;        // Used to convert returned value to have '0' as minimum. A leveller
+    float    sampleMax = 0.0f;    // Max sample over a few seconds. Needed for AGC controller.
```

Comment:
Veto. please don't change types without first consulting with me - some `double` vars are used intentionally.

Reply:
I tried to follow logic and it didn't seem to need that kind of precision. At least "not for a few lights that blink to a beat" as someone (bluntly) pointed out a while ago.
But I am not an expert and I will let you educate me.

Question: Do you *really* need precision on 8th decimal point? or are values in excess of 10^38?
Perhaps you do if you want great precision and do signal processing, but IMO there really is no need for that when "a few LEDs blink".

I understand quantization errors, but I still think that 7 decimal places might be enough for what we do with audio data in LED driver. Unless we a lot of looping while processing data. 7 digits is about 0.001‰ error in a single pass.

Do not get me wrong, I do not want to "prove" anything I'm just trying to understand if there really is a need for that kind of precision.

Reply:
Hi, for `sampleMax` it might be possible to go for less resolution. Actually I put this variable back to 'double' during my experiments 2 years ago, because I sometimes had very strange behaviour when the value is "float" and mic input is very noisy.

For `control_integrated` the case was very clear and double is needed. This is an integration value where up to 1000 small differences are added per second. So the integrator tends to "freak out" after some time, unless its double. I've seem this freakout happen, you suddenly get amplification values that bounce around between max and min.